src/aap_eda/api/serializers/activation.py

```diff
@@ -104,17 +105,24 @@ def _update_k8s_service_name(validated_data: dict) -> str:
     return service_name or create_k8s_service_name(validated_data["name"])
 
 
+def _extend_extra_vars_from_credentials(
```

Comment:
I see this function very confusing. `credential_data` is defined as `Union[str, int, Dict, List]` but it is expected to be a dictionary. The return type is defined as `Union[str, int, Dict, List]` but it can only return a dict or a string.

Reply:
updated.